src/Components/SingleArticle.jsx

```diff
@@ -8,27 +8,37 @@ import Card from '@mui/material/Card';
 import CardContent from '@mui/material/CardContent';
 import CardMedia from '@mui/material/CardMedia';
 import Typography from '@mui/material/Typography';
+import VoteButton from './VoteButton'
 
 export default function SingleArticle(){
 
   const { articleId } = useParams();
   const [article, setArticle] = useState(null);
   const [loading, setLoading] = useState(true);
   const [comments, setComments] = useState([]);
+  const [totalVotes, setTotalVotes] = useState(0);
 
   useEffect(() => {
     Promise.all([getArticleById(articleId), getCommentsById(articleId)])
     .then(([articleData, commentsData]) => {
       setArticle(articleData);
       setComments(commentsData);
       setLoading(false);
+
+      const commentsVotes = commentsData.reduce((sum, comment) => sum + comment.votes, 0)
+        setTotalVotes(articleData.votes + commentsVotes)
     })
+
     .catch((error) => {
       console.error("Error fetching data:", error);
       setLoading(false);
     });
 }, [articleId]);
 
+const handleVoteUpdate = (updatedVotes) => {
+  setTotalVotes((prevTotalVotes) => prevTotalVotes + updatedVotes)
```

Comment:
This is good for optimistically updating the votes but consider what should happen if the vote fails to get added to the db? We would need to undo this optimistic render :)

Reply:
This part scares me haha. I am just i the middle of completing my post comment. Once I have finished perhaps I will put in a help desk for the reverse opt rendering